src/edu/stanford/nlp/sempre/ParserState.java

```diff
@@ -227,6 +227,58 @@ protected void pruneCell(String cellDescription, List<Derivation> derivations) {
     return derivs;
   }
 
+  public List<Derivation> gatherTokenAndPhraseDerivations(int spanStart, int spanEnd) {
```

Comment:
Also, avoid duplicate code with previous function - refactor so things only show up once.  Probably the best is to have the above function call this one.